Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+                if (diff < currentDiff) {
+                    selectedFormat = format;
+                    currentDiff = diff;
+                }
+            }
+        }
+        NSError* error;
+
+        NSArray<AVCaptureDeviceFormat *> *depthFormats = selectedFormat.supportedDepthDataFormats;
+        NSArray<AVCaptureDeviceFormat *> *filtered = [depthFormats filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(AVCaptureDeviceFormat *evaluatedObject, NSDictionary<NSString *, id> *bindings) {
+            return CMFormatDescriptionGetMediaSubType(evaluatedObject.formatDescription) == kCVPixelFormatType_DepthFloat32 && CMVideoFormatDescriptionGetDimensions(evaluatedObject.formatDescription).width == 640 && CMVideoFormatDescriptionGetDimensions(evaluatedObject.formatDescription).height == 480;
+        }]];
+        AVCaptureDeviceFormat* depthFormat = nil;
+        if([filtered count] == 1){
+            depthFormat = filtered[0];
+        }
+        AVCaptureDeviceInput* videoDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:selectedDevice error:&error];
+        if(error != nil){
+            OTError *err = [OTError errorWithDomain:OT_PUBLISHER_ERROR_DOMAIN
+                                               code:OTCapturerError
+                                           userInfo:nil];
+            [self callDelegateOnError:err captureError:nil];
+            [self showCapturerError:err];
+            return;
+        }
+        [_capturer startWithDevice:selectedDevice format:selectedFormat sessionPreset:_capturePreset videoDeviceInput:videoDeviceInput videoMirrored:YES orientation:AVCaptureVideoOrientationPortrait pixelFormat:kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange depthFormat:depthFormat CompletionHandler:^(NSError * _Nullable error) {
+            if(error){
+                OTError *err = [OTError errorWithDomain:OT_PUBLISHER_ERROR_DOMAIN
+                                                   code:OTCapturerError
+                                               userInfo:nil];
+                [self callDelegateOnError:err captureError:nil];
+                [self showCapturerError:err];
+            }
+        }];
+    });
+    return 0;
+}
+
+- (int32_t)stopCapture {
+    _capturing = NO;
+    NSCondition *condition = [[NSCondition alloc] init];
+    [condition lock];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^(void){
+        [_capturer stopWithCompletionHandler:^{
+            [condition signal];
+        }];
+    });
+    [condition wait];
+    return 0;
+}
+
+- (BOOL)isCaptureStarted {
+    return _capturing;
+}
+
+- (int32_t)captureSettings:(OTVideoFormat*)videoFormat {
+    videoFormat.pixelFormat = OTPixelFormatNV12;
+    videoFormat.imageWidth = _captureWidth;
+    videoFormat.imageHeight = _captureHeight;
+    return 0;
+}
+
+- (void)updateCaptureFormatWithWidth:(uint32_t)width height:(uint32_t)height {
```

Comment:
i dont think this function is used.

Reply:
It might be called by the OpenTok video client SDK.

Reply:
Removed it.